### PR TITLE
TYP: Autotyping

### LINKED ISSUE #46337: TYP: returning None for __init__
xref https://github.com/pandas-dev/pandas/pull/44677#discussion_r771575699

we originally had a decision not to type annotate returning None for __init__

can other folks weigh in here? We have a number of PRs which do exactly this (I merged one or 2 then saw the comments)
e.g. https://github.com/pandas-dev/pandas/pull/46283 and a few others.

cc @simonjayhawkins @twoertwein @Dr-Irv @phofl

### PR DESCRIPTION
**Pull request by twoertwein**
Running autotyping on pandas https://github.com/JelleZijlstra/autotyping

Would be great to add it to pre-commit but before that many abstractmethods need return type annotations (autotyping let's them return `None` or wait for https://github.com/JelleZijlstra/autotyping/issues/33).

closes #46337
xref #47497

### CODE REVIEW

File: pandas/core/series.py [hunk 1]
```diff
@@ -1327,7 +1327,9 @@ def _maybe_update_cacher(
     # ----------------------------------------------------------------------
     # Unsorted
 
-    @property
+    # error: Cannot override final attribute "_is_mixed_type" (previously declared
+    # in base class "NDFrame")
+    @property  # type: ignore[misc]
     def _is_mixed_type(self) -> bool:
```
Comment (twoertwein) [inline pandas/core/series.py:1333]: `NDFrame` declares `_is_mixed_type` as final! This method should not exist or the parent method should not be final.

Not sure why mypy didn't find this earlier.

Comment (JelleZijlstra) [reply pandas/core/series.py:1333]: Mypy generally doesn't check unannotated methods. It should probably still check `@final` though.

**Comment by twoertwein** [comment]
This should be ready from my perspective @mroeschke Can work on integrating autotyping into the CI after its next release.

File: .pre-commit-config.yaml [hunk 2]
```diff
@@ -252,3 +252,14 @@ repos:
             /(__init__\.py)|(api\.py)|(_version\.py)|(testing\.py)|(conftest\.py)$
             |/tests/
             |/_testing/
+    -   id: autotyping
+        name: autotyping
+        entry: python -m libcst.tool codemod autotyping.AutotypeCommand --safe 1
```
[the remark this answers is not in the document]
Comment (twoertwein) [reply .pre-commit-config.yaml:257]: Okay, will change that.

Is it okay to add the `.libcst.codemod.yaml`? It is needed. If we don't want to clutter the main folder, I could create `scripts/run_autotyping.py` and create this file only temporarily.

[the remark this answers is not in the document]
Comment (mroeschke) [reply .pre-commit-config.yaml:257]: > Is it okay to add the .libcst.codemod.yaml?

Yeah sure that would be okay!

**Review by mroeschke — APPROVED** [review]

**Comment by mroeschke** [comment]
Thanks @twoertwein!

**Pull request merged**